libmisc/addgrps.c

```diff
@@ -88,14 +88,12 @@ int add_groups (const char *list)
 			fputs (_("Warning: too many groups\n"), shadow_logfd);
 			break;
 		}
-		tmp = (gid_t *) reallocarray (grouplist, (size_t)ngroups + 1, sizeof (GETGROUPS_T));
-		if (NULL == tmp) {
-			free (grouplist);
+		grouplist = (gid_t *) reallocarrayf (grouplist, (size_t)ngroups + 1, sizeof (GETGROUPS_T));
```

Comment:
I'm not terribly worried about this, but just thought I'd point out:  reallocarrayf on failure will call free(), and I see that free not setting errno on failure is promised in the future, it is not right now https://sourceware.org/bugzilla/show_bug.cgi?id=17924 .  So maybe reallocarrayf() should save and re-set errno.

Reply:
Hmm, that bug seems outdated.  The manual page was updated (I didn't check but I guess it was also @eggert in the same patch mentioned earlier) to say:

```
       The free() function returns no value, and preserves errno.

...

       free() preserved errno starting in glibc 2.33.
```

Paul, could you close the bug ticket?

Reply:
My bad; it seems already closed: 

```
Status: RESOLVED FIXED
```